### DISCUSSION TITLE
Improving documentation

### DISCUSSION THREAD

**Discussion by AndersRask** [pnp/powershell#1414] [General]
Hi, I wanted to add to the documentation md for some of the cmdlets, but is a bit confused regarding weather I need to do a complete fork/PR every time, or if there is an easier way to do it?

On the Contributing.md file it states

> Documentation contributions
> The PowerShell documentation is located on https://docs.microsoft.com/en-us/powershell/sharepoint/sharepoint-pnp/sharepoint-pnp-cmdlets?view=sharepoint-ps
https://github.com/pnp/powershell/blob/dev/CONTRIBUTING.md

And there is an Edit mode for that page, but on every cmdlet it say "please navigate to xxx.md to change this file"
![image](https://user-images.githubusercontent.com/1651150/144573892-a7df84f9-5150-4bb8-9ed8-43d508591bc8.png)

On said page, when you edit it, you get a warning that you need to create a fork.

I guess a fork is the way to go, but the barrier for getting your average user to improve documentation is probably very low with this setup :)

Any thoughts?

**Comment by LeonArmston** [comment]
Hi 

I recommend you join one of the great PnP sharing is caring sessions run by the PnP Team mainly @PopWarner and @hugoabernier ! These great guys will walk you around GitHub and introduce all the concepts to you in no time and before you know it you will submitting pull requests to contribute to PnP.

I recommend you join upto a First Time Contributor session (see link below)
https://pnp.github.io/sharing-is-caring/#pnp-sic-events

Any questions dont hesitate to reach out!

**Reply by AndersRask** [reply]
Hi Leon,
I do dev work in GitHub or DevOps every day, so for me the issue is mostly that the guidance on how to add to documentation is not very clear (as described in the post). 
The issue here is, that many users of PnP.PowerShell is IT-pros or similar, or consultants doing quick fixes etc. They will never fork a repo to add documentation, but they use the cmdlets a lot, and have valuable knowledge around what works and what does not, so making it easier for people to update documentation directly in GitHub would reduce the barrier for those people greatly and you would start seeing actual updates to documentation

**Reply by LeonArmston** [reply]
Fair enough - sounds like you know your way around GitHub which is great! Still recommend the sharing is caring sessions as they will show you the PnP way of contributing and introduce you to some of the team.

The way I do it if it helps and I'm not sure if you can do it any other way as we dont have admin rights to the source repo:

- Fork the repo to my own GitHub
- Create branch for the changes I want to complete
- Make the changes to the files that I want to complete in my branch.
- Then raise a pull request - posting as much detail in the pull request as necessary.
- Wait for one of the PnP PowerShell team to review and if ok merge into the source repo.
